src/commons.rs

```diff
@@ -246,53 +166,125 @@ pub extern "C" fn z_qos_default() -> z_qos_t {
 ///   z_attachment_t attachment: The attachment of this data sample.
 #[repr(C)]
 pub struct z_sample_t<'a> {
-    pub keyexpr: z_keyexpr_t,
-    pub payload: z_bytes_t,
-    pub encoding: z_encoding_t,
-    pub _zc_buf: &'a c_void,
-    pub kind: z_sample_kind_t,
-    pub timestamp: z_timestamp_t,
-    pub qos: z_qos_t,
-    pub attachment: z_attachment_t,
+    _inner: &'a (),
+}
+impl<'a> core::ops::Deref for z_sample_t<'a> {
+    type Target = Sample;
+    fn deref(&self) -> &Self::Target {
+        unsafe { core::mem::transmute::<&(), &Sample>(self._inner) }
+    }
 }
 
 impl<'a> z_sample_t<'a> {
-    pub fn new(sample: &'a Sample, owner: &'a ZBuf) -> Self {
-        let std::borrow::Cow::Borrowed(payload) = owner.contiguous() else {
+    pub fn new(sample: &'a Sample) -> Self {
+        if !sample.value.payload.zslices().count() <= 1 {
```

Comment:
`if ! zslices.count <= 1 `is exactly `if zslices.count >1`, but less readable
And should we also check for emply payload: `zslices.count == 0` ?

Reply:
With the exposure of ZBuf, we actually don't want to check it at all anymore, because samples with non-contiguous payloads are now allowed